### PR TITLE
Relaxing owner connection check for client invocations

### PR DESCRIPTION
**Pull request by sancar**
Owner connection check is only needed for the invocations that
will acquire resources like map.lock/trylock lock.lock/trylock
and most of the semaphore operations. Other invocations does not
need owner connection to be available. This improvement is made
to isolate owner connection failure from invocations on other
connections.

### CODE REVIEW

**Comment by sancar** [comment]
@ihsandemir I have just added a test. Can you recheck the pr?

File: hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/AbstractClientInvocationService.java [hunk 1]
```diff
@@ -117,9 +117,13 @@ protected void send(ClientInvocation invocation, ClientConnection connection) th
         if (isShutdown) {
             throw new HazelcastClientNotActiveException("Client is shut down");
         }
+        ClientMessage clientMessage = invocation.getClientMessage();
+        if (clientMessage.isAcquiresResource() && connectionManager.getOwnerConnection() == null) {
```
Comment (pveentjer) [inline hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/AbstractClientInvocationService.java]: I would name this to hasAquiredResource.

Comment (sancar) [reply hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/AbstractClientInvocationService.java]: `acquiresResource` sounds better to me too. I am changing it.

File: hazelcast-client/src/main/java/com/hazelcast/client/proxy/txn/TransactionProxy.java [hunk 2]
```diff
@@ -70,6 +70,9 @@ public State getState() {
 
     void begin() {
         try {
+            if (client.getConnectionManager().getOwnerConnection() == null) {
+                throw new TransactionException("Owner connection needs to be present to begin a transaction");
```
Comment (ihsandemir) [inline hazelcast-client/src/main/java/com/hazelcast/client/proxy/txn/TransactionProxy.java]: Can you add tests for such transactıon exceptions?

Comment (sancar) [reply hazelcast-client/src/main/java/com/hazelcast/client/proxy/txn/TransactionProxy.java]: I have added a new test for  transactions `ClientTxnOwnerDisconnectedTest`

**Comment by sancar** [comment]
run-lab-run

**Pull request merged**